index.bs

```diff
@@ -892,7 +896,70 @@ option. If {{UnderlyingSource/type}} is set to undefined (including via omission
  </div>
 </div>
 
-<!-- TODO async iterator stuff -->
+<h4 id="rs-asynciterator" oldids="rs-asynciterator-prototype,
+default-reader-asynciterator-prototype-internal-slots">Asynchronous iteration</h4>
+
+<dl class="domintro">
+ <dt><code>for await (const <var ignore>chunk</var> of <var ignore>stream</var>) { ... }</code>
+ <dt><code>for await (const <var ignore>chunk</var> of <var ignore>stream</var>.values([{ {{ReadableStreamIteratorOptions/preventCancel}} }])) { ... }</code>
```

Comment:
I think the square brackets are a typo? You should be able to call this as `.values({ preventCancel })`, correct?
```suggestion
 <dt><code>for await (const <var ignore>chunk</var> of <var ignore>stream</var>.values({ {{ReadableStreamIteratorOptions/preventCancel}} })) { ... }</code>
```

Reply:
They indicate optionality, not array-ness. I can see how it's confusing though :-/. Maybe in this case at least we could just do `{ preventCancel: true }` to show the two different styles.

Reply:
Ooooh right, I see it know. There's a similar thing going on with `reader = stream.getReader([{ mode: "byob" }])`, but for some reason I didn't interpret that as an array. I think seeing the lengthy `for await of` construct around it made me interpret the whole thing as a line of JavaScript, rather than a signature with optional arguments.

It seems like the syntax used in these non-normative sections is a bit ambiguous, since we use square brackets for both optionality and arrays:
* `stream = new ReadableStream(underlyingSource[, strategy])` (optionality)
* `[branch1, branch2] = stream.tee()` (array return value)

Then again, I can't immediately come up with a better way to present this. 🤷 
(Something like `{ preventCancel }?` would work in TypeScript, but that's not a thing in JavaScript.)